### DISCUSSION TITLE
Strange screen position jumping behavior after surrounding with function

### DISCUSSION THREAD

**Discussion by kietcthe** [JetBrains/ideavim#609] [Q&A]
Hi,
I want to ask if anyone here has faced this problem with the surround feature.
The problem is like in the below video:

https://user-images.githubusercontent.com/125039563/222870248-c8460168-8eca-4e10-b78b-1620f1fc912e.mov

**Step-by-step procedure:**
- Select text in visual mode.
- Type S, then type f to use the surround with function feature.
- Type the name of the function. Although I typed the widget name in the video, I think it's the same as the function name.

**Expect:** The screen position stays at the current line after surround.
**Actually:** The cursor still stays at the current line, but the screen position jumps away.

My .ideavimrc file content:
`set surround`

I would appreciate it if anyone could help me with this strange behavior.

**Comment by ljchen1** [comment]
Sadly, I can't reproduce your issue. Try to have `set surround` only in the `.ideavimrc` to see if you could reproduce this issue.
if still have this problem, would it possible to share a sample project for reproducing this issue?

**Reply by kietcthe** [reply]
Thank you for taking the time to assist me.
And yes, my .ideavimrc file only has 1 line: set surround.
You'll see this strange behavior when you wrap the string 'You have pushed the button this many times' with a Text widget, like in the image below.
<img width="1655" alt="image" src="https://user-images.githubusercontent.com/125039563/223983376-a7319217-6e5c-4f32-91a0-f9cffa2ba93a.png">

Here is the sample project.
[Counter.zip](https://github.com/JetBrains/ideavim/files/10929963/Counter.zip)

**Reply by ljchen1** [reply]
Your example works correctly in my local too.
Try to disable all downloaded plugins from IDE main menu `Settings/Preferences | Plugins | "Installed" Tab` except the IdeaVIM plugin and restart the IDE to see whether a plugin causes this problem. (You could reenable them all together in the same place)


https://user-images.githubusercontent.com/103916765/224309506-5e87a913-ae9c-41e7-bc34-ddfe607e7df0.mp4

**Reply by kietcthe** [reply]
<img width="742" alt="image" src="https://user-images.githubusercontent.com/125039563/224321500-51cb6e30-fecc-468e-aaba-044bf89bbe1c.png">
I tried, but it still occasionally appears (though not always). Just try wrapping other place texts. I think you will encounter this behavior.

**Reply by ljchen1** [reply]
Thanks. I could reproduce it now after trying it 2-3 times.
I have created a bug report here and you could vote for it for updates: https://youtrack.jetbrains.com/issue/VIM-2895/

**Reply by kietcthe** [reply]
Thank you very much 🤲
